src/Classes/LinkRenderers/GenericLinkRenderer.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace LaraZeus\Sky\Classes\LinkRenderers;
+
+use Illuminate\Database\Eloquent\Model;
+
+class GenericLinkRenderer extends NavLinkRenderer
+{
+    public function getModel(): ?Model
```

Comment:
The generic Renderer doesn't need this method technically.
Totally possible to do if desired, but didn't want to over-engineer.

Essentially we would create a new `abstract class NavLinkWithModelRenderer extends NavLinkRenderer` to use with types that are model based. Which could be a great DX, but unsure if worth it since we can support both by making it nullable like I have here.

Reply:
np for now, we can always go back and improve.